apps/settings/js/modules/dialog_manager.js

```diff
@@ -177,12 +177,21 @@ define(function(require) {
         return self._getPanel(dialog.panel.id);
       })
       .then(function(panel) {
-        // 4: call beforeShow
+        // 4: Inject cancel(), submit() functions for panel use
```

Comment:
If the operation is synchronous, it is fine to put them in the existing function and use the comments to specify the step.